sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -237,8 +237,10 @@ case class DropTableCommand(
 
     if (isTempView || catalog.tableExists(tableName)) {
       try {
+        val hasViewText = isTempView &&
+          catalog.getTempViewOrPermanentTableMetadata(tableName).viewText.isDefined
         sparkSession.sharedState.cacheManager.uncacheQuery(
-          sparkSession.table(tableName), cascade = !isTempView)
+          sparkSession.table(tableName), cascade = !isTempView || hasViewText)
```

Comment:
Is the PR title and description up-to-dated? Looks like the change in the PR is basically changing the uncaching behavior of view with view text.

Looks like we already did cascade uncache when dropping a source table here, so I'm wondering why the title is "A cached view should become invalid after the source table is dropped"?

Reply:
Yes the PR title is not very good - I just updated it. I think the description is more aligned to what the PR does but I'll try to add more details there.